src/domain/dataset-update-flow/src/entities/update/update_state.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use event_sourcing::*;
+use kamu_task_system::{TaskID, TaskOutcome};
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+const MAX_UPDATE_TASKS: usize = 3;
```

Comment:
Retries are not yet implemented. It can be done in different ways, and one of them is to have a domain-level constraint, as the state sees all previously schedule task IDs.